### PR TITLE
Reformat the insights report to contain smaller slices

### LINKED ISSUE #1909: Modify insights endpoint to break large reports into multiple reports and return them all along with a metadata payload
## Feature description:
To improve yupana performance, we want to ensure that large reports are broken down into multiple reports, as well as display the metadata in an easily loadable way. We should modify the task & insights endpoint to break large reports into chunks and generate a metadata section. 

### Is your feature request related to a problem?
Insights reports can be very large. Additionally, we can't validate a report without loading the JSON. The act of loading a very large JSON file could cause our service to have memory issues.

Since we are sending a tar.gz to the file upload service, we should split out the metadata into a small JSON payload that will always load and a report JSON payload (possibly multiple report payloads).

### Describe the solution you'd like
There should be no more than 10k hosts per report. Reports that have higher than this max should be split out in a logical way (a report with 10,001 should not be two reports 10k and 1 but a 20k system report should be 10k & 10k). A metadata section should also be generated that has the following info:
```
{"report_id": "x",
 "report_version": "y",
 "qpc_server_version": "foo",
 "qpc_server_id": "bar",
 "slices": {"x-1": {"number_host": 5, "state": NEW}}
 }
```

Additionally, we should remove the insights report from the `reports/id` endpoint 
___

## Acceptance Criteria:

- [ ] Verify that large reports are broken up 
- [ ] Verify that a metadata payload is generated 

___

### PR DESCRIPTION
**Pull request by kholdaway**
Closes #1909

### CODE REVIEW

File: quipucords/api/insights_report/view.py [hunk 1]
```diff
@@ -68,96 +71,70 @@ def insights(request, pk=None):
...
-    return insights_hosts
-
-
-def build_cached_insights_json_report(report):
-    """Create an insights report based on a deployments report.
-
-    :param report: DeploymentsReport that is used to create insights report
-    :returns: json report data
-    :raises: Raises failed dependencies if no hosts have canonical facts
+def slice_it(host_key_list, hosts_per_slice):
```
Comment (abaiken) [inline quipucords/api/insights_report/view.py]: This is cool.

File: quipucords/api/insights_report/insights_gzip_renderer.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Copyright (c) 2018-2019 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""tar.gz renderer for reports."""
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply quipucords/api/insights_report/insights_gzip_renderer.py]: Done!

File: quipucords/api/insights_report/insights_gzip_renderer.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply quipucords/api/insights_report/insights_gzip_renderer.py]: Thanks!

File: quipucords/api/insights_report/insights_gzip_renderer.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+from api.common.common_report import (create_tar_buffer)
+
+from rest_framework import renderers
+
+# Get an instance of a logger
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class InsightsGzipRenderer(renderers.BaseRenderer):
+    """Class to render all reports as tar.gz."""
```
[the remark this answers is not in the document]
Comment (kholdaway) [reply quipucords/api/insights_report/insights_gzip_renderer.py]: 🌮

File: quipucords/api/insights_report/view.py (hunk 1 above)
Comment (kholdaway) [reply quipucords/api/insights_report/view.py]: 😭 I know

**Pull request merged**